pmd-core/src/main/java/net/sourceforge/pmd/processor/PmdRunnable.java

```diff
@@ -82,6 +83,9 @@ public Report call() {
 
         try (InputStream stream = new BufferedInputStream(dataSource.getInputStream())) {
             tc.ruleContext.setLanguageVersion(null);
+            if (dataSource instanceof LanguageAwareDataSource) {
+                tc.ruleContext.setLanguageVersion(((LanguageAwareDataSource) dataSource).getLanguageVersion());
```

Comment:
Setting the language here prevents us from trying to determine the language from the file extension later.